Add batch invariant RL post

bwasti opened the pull request:
As in title, text can be found in PR content

bwasti committed:
Create 2025-11-10-bitwise-exact-rl.md

Signed-off-by: Bram Wasti <bwasti@fb.com>

bwasti committed:
Add links to batch invariance and RFC in documentation

Signed-off-by: Bram Wasti <bwasti@fb.com>

bwasti committed:
Fix broken images in bitwise-exact RL blog post

Downloaded GitHub-hosted images to local assets directory and updated
all image references to use local paths. Converted standalone images to
markdown syntax while keeping centered images as HTML img tags for
proper rendering.

Signed-off-by: Bram Wasti <bwasti@meta.com>

bwasti committed:
Add frontmatter to bitwise-exact RL blog post

Added Jekyll frontmatter with layout, title, and author metadata to
properly render the blog post.

Signed-off-by: Bram Wasti <bwasti@meta.com>

bwasti committed:
Restore width and height attributes for centered images

Restored the original width and height attributes (340x130 and 480x355)
for the two centered images to maintain their fixed sizing.

Signed-off-by: Bram Wasti <bwasti@meta.com>

bwasti committed:
Format acknowledgements section

Added horizontal rule and italic formatting to the acknowledgements
section for better visual separation and styling.

Signed-off-by: Bram Wasti <bwasti@meta.com>

zhuohan123 commented on _posts/2025-11-10-bitwise-consistent-train-inference.md:
```diff
@@ -0,0 +1,59 @@
+
+---
+*Acknowledgements:
+Bram Wasti, Teja Rao, Paul Zhang, Tianyu Liu, Zhuohan Li, Natalia Gimelshein*
```
```suggestion
## Acknowledgements

Blog written by Bram Wasti. We would like to thank the team from vLLM and TorchTitan who contributed to this effort: Bram Wasti, Teja Rao, Paul Zhang, Tianyu Liu, Zhuohan Li, Natalia Gimelshein.
```

mgoin replied:
Please add Wentao Ye as well!

WoosukKwon commented on _posts/2025-11-10-bitwise-exact-rl.md:
```diff
@@ -0,0 +1,59 @@
+
+In the septillions of flops used to pre-train models, this mismatch between values has largely been avoidable.  Pre-training typically runs at a fixed batch size which induces the same reduction kernels to be run - often side-stepping the issue entirely.
+
+Reinforcement learning, on the other hand, seems to almost exclusively run different reduction algorithms due to its inference-heavy (and thus largely latency and memory-bound) nature.  Kernels optimized for low-batch size inference typically run reductions all at once, whereas kernels for training models parallelize heavily to reuse data and amp up compute utilization.  That means the generators ...
```
> Kernels optimized for low-batch size inference typically run reductions all at once

I don't understand this part. Are you talking about reductions like in RMS norm?

bwasti replied:
in the kernels, they don't tile. let me use the word "tile" for clarity

youkaichao commented on _posts/2025-11-10-bitwise-exact-rl.md:
```diff
@@ -0,0 +1,59 @@
+---  
+layout: post
+title: "Bitwise-exact Batch Invariant On-Policy Reinforcement Learning with vLLM and TorchTitan"
```
suggested title:

`No More Train/Inference Mismatch: Bitwise Consistent On-Policy Reinforcement Learning with vLLM and TorchTitan`

youkaichao commented on _posts/2025-11-10-bitwise-consistent-train-inference.md:
filename will become the blogpost url. we can make the url more attractive like `2025-11-10-bitwise-consistent-train-inference.md`

youkaichao commented on _posts/2025-11-10-bitwise-exact-rl.md:
```diff
@@ -0,0 +1,59 @@
+author: "vLLM  and TorchTitan Teams"   
+---
+
+In the septillions of flops used to pre-train models, this mismatch between values has largely been avoidable.  Pre-training typically runs at a fixed batch size which induces the same reduction kernels to be run - often side-stepping the issue entirely.
```
what is "this mismatch"? do we miss any paragraphs?

bwasti replied:
oops, some paragraph reorganization was done (we have a couple people editing this in a google doc elsewhere). will fix

bwasti committed:
Update 2025-11-10-bitwise-exact-rl.md

Signed-off-by: Bram Wasti <bwasti@fb.com>

bwasti committed:
Rename bitwise-exact-rl.md to bitwise-consistent-train-inference.md

Signed-off-by: Bram Wasti <bwasti@fb.com>

bwasti committed:
Update authors in bitwise consistent train inference post

Signed-off-by: Bram Wasti <bwasti@fb.com>

bwasti committed:
Update 2025-11-10-bitwise-consistent-train-inference.md

Signed-off-by: Bram Wasti <bwasti@fb.com>

zhuohan123 committed:
zhuohan's rewrote on bitwise blog

bwasti committed:
Merge pull request #1 from vllm-project/zhuohan/bitwise-rewrote

bwasti committed:
Update 2025-11-10-bitwise-consistent-train-inference.md

Signed-off-by: Bram Wasti <bwasti@fb.com>

youkaichao committed:
Merge branch 'main' into patch-1

bwasti committed:
Add Slack channels for contributing to bitwise consistency work

Added links to #sig-post-training and #sig-batch-invariant Slack channels
in the blog post to invite readers to contribute to future developments.

Signed-off-by: Bram Wasti <bwasti@meta.com>
